src/Hyperion/Benchmark.hs

```diff
@@ -55,7 +60,37 @@ instance Show Benchmark where
   showsPrec x (WithSampling _opt bk) = showsPrec x bk
 
 bench :: String -> Batch () -> Benchmark
-bench name batch = Bench (Text.pack name) batch
+bench name batch = Bench Text.pack (const []) id (const batch) name
+
+benchWith
+  :: (a -> Text)
+  -> (a -> [JSON.Pair])
+  -> (a -> b)
+  -> (b -> Batch ())
+  -> a -> Benchmark
+benchWith = Bench
+
+benchMany
```

Comment:
I don't understand the use case for `benchMany`. It looks to me like you're reinventing an existing functionality in Hyperion, namely "series". Could you clarify how the two differ?

The idea with series is precisely to support one of the use cases you mention in your description, namely that you have a set of benchmarks indexed e.g. by input size. So then you're free to plot time against input size if you can get Kibana to do that.